### ISSUE TITLE
Gibberish logs with TCP DNS

### ISSUE THREAD

**Issue by klzgrad** [SagerNet/SagerNet#128]
Reproduce: Local DNS: tcp://8.8.8.8:53, and some other steps I forgot.

logcat actual
```
GoLog   : 127.0.0.1:52356 accepted udp:tcp://8.8.8.8:53:53 [dns-out]
GoLog   : Not Using Prepared: tcp,8.8.8.8:53
GoLog   : 127.0.0.1:40729 accepted udp:tcp://8.8.8.8:53:53 [dns-out]
GoLog   : Not Using Prepared: tcp,8.8.8.8:53
GoLog   : Not Using Prepared: udp,tcp://8.8.8.8:53:53
GoLog   : 127.0.0.1:37420 accepted udp:tcp://8.8.8.8:53:53 [dns-out]
GoLog   : PrepareDomain SplitHostPort Err: address tcp://8.8.8.8:53:53: too many colons in address
GoLog   : [Warning] proxy/dns: failed to dial outbound connection > address tcp://8.8.8.8:53:53: too many colons in address
GoLog   : Not Using Prepared: udp,tcp://8.8.8.8:53:53
GoLog   : 127.0.0.1:32967 accepted udp:tcp://8.8.8.8:53:53 [dns-out]
GoLog   : PrepareDomain SplitHostPort Err: address tcp://8.8.8.8:53:53: too many colons in address
GoLog   : [Warning] proxy/dns: failed to dial outbound connection > address tcp://8.8.8.8:53:53: too many colons in address
GoLog   : Not Using Prepared: udp,tcp://8.8.8.8:53:53
GoLog   : PrepareDomain SplitHostPort Err: address tcp://8.8.8.8:53:53: too many colons in address
GoLog   : [Warning] proxy/dns: failed to dial outbound connection > address tcp://8.8.8.8:53:53: too many colons in address
GoLog   : Not Using Prepared: udp,tcp://8.8.8.8:53:53
GoLog   : PrepareDomain SplitHostPort Err: address tcp://8.8.8.8:53:53: too many colons in address
GoLog   : [Warning] proxy/dns: failed to dial outbound connection > address tcp://8.8.8.8:53:53: too many colons in address
GoLog   : 127.0.0.1:36653 accepted udp:tcp://8.8.8.8:53:53 [dns-out]
```

logcat expected:
* Things like `udp:tcp://8.8.8.8:53:53` do not appear.
* Things like `address tcp://8.8.8.8:53:53: too many colons in address` do not appear.

**Comment by AkinoKaede** [comment]
The configures that SagerNet generated is wrong...

**Comment by nekohasekai** [comment]
Should be fixed in [996f72a](https://github.com/SagerNet/SagerNet/commit/996f72a7aa93f615c3f4880aacd00a94d214e850)

**Comment by klzgrad** [comment]
Probably verified.
